### DISCUSSION TITLE
Add a `useHistory` to solid-router

### DISCUSSION THREAD

**Discussion by Bersaelor** [solidjs/solid-router#290] [Ideas]
In [react-router we had a `useHistory` hook](https://v5.reactrouter.com/web/api/Hooks/usehistory) that allows us to base logic on our current site on the route that the user navigated through the SPA to the current location.

`useLocation` gives us the current location but often it would be helpful to know the history how the user got to it :)

**Comment by Bersaelor** [comment]
As a workaround, in my root.tsx, I now do:

```js

declare global {
  interface Window {
    solid_history?: string[]
  }
}

export default function App() {

  onMount(() => {
    document.body.style.background = '#3F565E';
  });

  createEffect(() => {
    // subscribe to the location and push it on the history stack
    const path = useLocation().pathname;
    console.log('location changed: ', path);
    if (window.solid_history) {
      window.solid_history.push(path);
    } else {
      window.solid_history = [path];
    }
  });
  
    // ...
  ```
  
  which I can then read in pages as:
  ```js
    const solid_history = window.solid_history;
    const prevPath = solid_history && solid_history.length > 1 && solid_history[solid_history.length - 2];
```

**Reply by aseerkt** [reply]
I think you can use `useNavigate` from `@solidjs/router`
And also In react-router-dom v6 - `useHistory` is replace with `useNavigate`
